### PR TITLE
Implement Task2

### PR DESCRIPTION
**Pull request by kundro**

### CODE REVIEW

**Comment by kundro** [comment]
"React navigation" commit:
1) I installed the libraries like @react-navigation/native, react-native-screens and react-native-safe-area-context.
2) Wrapped my app in NavigationContainer. I'll try to do it inside index.js or create new file named App.js.
3) Created a native stack navigator and configured the navigation.
4) Added a second screen to the native stack navigator (with information about me).
5) Added the button named 'Profile' and configured the navigating to AboutMe.screen.js
Screenshot 2022-05-12 151451
Screenshot 2022-05-12 151520

File: AwesomeProject/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import {AppRegistry} from 'react-native';
 import AboutMe from './screens/AboutMe/AboutMe.screen';
+import Screens from './screens/Menu/Menu.screen';
 import {name as appName} from './app.json';
 
-AppRegistry.registerComponent(appName, () => AboutMe);
+AppRegistry.registerComponent(appName, () => Screens, AboutMe);
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/index.js]: you don't need AboutMe here, remove it from register and from importing here

File: AwesomeProject/screens/AboutMe/AboutMe.screen.js [hunk 2]
```diff
@@ -2,31 +2,29 @@ import React from 'react';
...
-            thought it would be great for me. But also i have a good logical
-            thinking.</Text> 
-            <Text style={styles.mainText}>Contacts: exey.kundro@gmail.com</Text>
-        </View>
-      </ScrollView>
-    </View>
-  );
-};
+AboutMe = () => (
+  <View style={styles.container}>
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/AboutMe/AboutMe.screen.js]: no View component here. Move style to ScrollView

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import {createNativeStackNavigator} from '@react-navigation/native-stack';
+import {NavigationContainer} from '@react-navigation/native';
+import {Button, View, ScrollView} from 'react-native';
+import styles from '../Menu/Menu.styles';
+import AboutMe from '../AboutMe/AboutMe.screen';
+
+export default Screens = () => (
+  <NavigationContainer>
+    <Stack.Navigator initialRouteName="Screens">
+      <Stack.Screen name="Menu" component={MenuScreen} />
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: create 'config' folder on the same level as 'screens' and create constants.js file.
MENU_SCREEN and ABOUT_ME_SCREEN and import them here and pass to name property.

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import {createNativeStackNavigator} from '@react-navigation/native-stack';
+import {NavigationContainer} from '@react-navigation/native';
+import {Button, View, ScrollView} from 'react-native';
+import styles from '../Menu/Menu.styles';
+import AboutMe from '../AboutMe/AboutMe.screen';
+
+export default Screens = () => (
+  <NavigationContainer>
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: navigation should be in nav.js file. This file should be inside nav folder on the same level with 'screens' folder.

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    <Stack.Navigator initialRouteName="Screens">
+      <Stack.Screen name="Menu" component={MenuScreen} />
+      <Stack.Screen name="Alexey Kundro" component={DetailsScreen} />
+    </Stack.Navigator>
+  </NavigationContainer>
+);
+
+MenuScreen = ({navigation}) => (
+  <View style={styles.container}>
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: remove this View and pass style to ScrollView

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    <ScrollView>
+      <View style={styles.main}>
+        <Button
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: This is simple button that don't have any styles and it doesn't suit us. For adding button like we discussed find another components in react-native for this.

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import {NavigationContainer} from '@react-navigation/native';
+import {Button, View, ScrollView} from 'react-native';
+import styles from '../Menu/Menu.styles';
+import AboutMe from '../AboutMe/AboutMe.screen';
+
+export default Screens = () => (
+  <NavigationContainer>
+    <Stack.Navigator initialRouteName="Screens">
+      <Stack.Screen name="Menu" component={MenuScreen} />
+      <Stack.Screen name="Alexey Kundro" component={DetailsScreen} />
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: component={AboutMe} !!!!!!

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+);
+
+MenuScreen = ({navigation}) => (
+  <View style={styles.container}>
+    <ScrollView>
+      <View style={styles.main}>
+        <Button
+          style={styles.buttonProfile}
+          title="PROFILE"
+          onPress={() => navigation.navigate('Alexey Kundro')}
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: on this navigation you need to pass constant ABOUT_ME

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        />
+      </View>
+    </ScrollView>
+  </View>
+);
+
+DetailsScreen = ({navigation}) => AboutMe();
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: delete it

File: AwesomeProject/package.json [hunk 10]
```diff
@@ -10,8 +10,11 @@
     "lint": "eslint ."
   },
   "dependencies": {
+    "@react-navigation/native-stack": "^6.6.2",
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/package.json:14]: you haven't installed library with icons

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: import React from 'react' is enough

File: AwesomeProject/screens/Menu/Menu.screen.js (hunk 7 above)
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: if you want to pass you name to the title use options property for the screen. Find more information about it inside react-navigation api for StackNavigator

File: AwesomeProject/nav/nav.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+import {NavigationContainer} from '@react-navigation/native';
+import {createStackNavigator} from '@react-navigation/stack';
+
+const Stack = createStackNavigator();
+
+export default Navigate = () => (
+  <NavigationContainer>
+    <Stack.Navigator>
+      <Stack.Screen
+        name="MENU_SCREEN"
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/nav/nav.js]: export const MENU_SCREEN = 'menu';
export const ABOUT_ME_SCREEN = 'about_me';

File: AwesomeProject/nav/nav.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import AboutMe from '../screens/AboutMe/AboutMe.screen';
+import Menu from '../screens/Menu/Menu.screen';
+
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/nav/nav.js]: import { MENU_SCREEN, ...} from '../config/constants';

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import {Text, TouchableOpacity, View, ScrollView} from 'react-native';
+import styles from '../Menu/Menu.styles';
+
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js:4]: import { ABOUT_ME_SCREEN} from '../../config/constants';

File: AwesomeProject/screens/AboutMe/AboutMe.screen.js [hunk 15]
```diff
@@ -2,31 +2,25 @@ import React from 'react';
...
-            thinking.</Text> 
-            <Text style={styles.mainText}>Contacts: exey.kundro@gmail.com</Text>
-        </View>
-      </ScrollView>
-    </View>
-  );
-};
-
-export default AboutMe;
+export default AboutMe = () => (
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/AboutMe/AboutMe.screen.js]: const AboutMe = () => ....

export default AboutMe;

File: AwesomeProject/index.js [hunk 16]
```diff
@@ -1,5 +1,5 @@
 import {AppRegistry} from 'react-native';
-import AboutMe from './screens/AboutMe/AboutMe.screen';
 import {name as appName} from './app.json';
+import Menu from './screens/Menu/Menu.screen';
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/index.js:4]: why you doing this? you need to import here nav.js  :)

File: AwesomeProject/screens/Menu/Menu.screen.js [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+export default function Menu({navigation}) {
+  const loadScene = () => {
+    navigation.navigate('ABOUT_ME_SCREEN');
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.screen.js]: use ABOUT_ME_SCREEN constant

File: AwesomeProject/screens/Menu/Menu.styles.js [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+    borderRadius: 5,
+    borderColor: colors.dimgrey
+
+  },
+  mainText: {
+    padding: 7,
+    fontSize: 18,
+    color: colors.text,
+    fontFamily: 'Roboto',
+    // fontWeight: '',
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/screens/Menu/Menu.styles.js]: remove unused line

**Comment by kundro** [comment]
Changed almost all files. Accordingly, the result differs from the first PR.

https://user-images.githubusercontent.com/76398633/172496824-3ef638e1-59fa-486d-ae22-4243febbe257.mp4

Task2_2
tast2_1

File: AwesomeProject/styles/AboutMe.styles.js [hunk 19]
```diff
@@ -1,18 +1,17 @@
 import {StyleSheet} from 'react-native';
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/styles/AboutMe.styles.js]: styles should be in the same place with file where it used. In your case this file should be in AboutMe folder

File: AwesomeProject/styles/Button.styles.js [hunk 20]
```diff
@@ -0,0 +1,23 @@
+import {StyleSheet} from 'react-native';
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/styles/Button.styles.js]: This is should be with Button file

File: AwesomeProject/Components/Button.js [hunk 21]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (AliakseiPaseishvili) [inline AwesomeProject/Components/Button.js]: This file should be in Button folder

Comment (AliakseiPaseishvili) [reply AwesomeProject/Components/Button.js]: styles for this file should in the same folder

**Pull request merged**